common/types/ballot.go

```diff
@@ -58,11 +58,19 @@ type Ballot struct {
 	// the proof of the smeshers eligibility to vote and propose block content in this epoch.
 	// Eligibilities must be produced in the ascending order.
 	// the proofs are vrf signatures and need not be included in the ballot's signature.
-	// according to protocol there are 50 per layer, the rest is safety margin
-	EligibilityProofs []VotingEligibility `scale:"max=500"`
+	//
+	// The number of eligibility proofs depends on the smeshers weight and the total weight of the network.
+	// For epoch 16 the largest smesher had 888 SUs and the total weight of the network was ~20.2 Mio SUs.
+	// This means that the largest smesher received 888 / 20,200,000 = 0.0044% of all eligibility slots for the epoch.
+	// There are 4032 layers in an epoch and 50 eligibility slots per layer, so the largest smesher received
+	// 0.0044% * 4032 * 50 = ~9 eligibility slots.
+	//
+	// Assuming the largest smesher won't control more than 10% of space in the network, we can assume that the
+	// highest number of eligibilities in a single ballot will be below 25000. (10% of 4032 * 50 = 20160)
+	EligibilityProofs []VotingEligibility `scale:"max=25000"`
```

Comment:
we are not concerned about number of eligibility proofs for a single identity.
instead number of identities will grow, and therefore number of proposals/ballots. so we have to update collections that references proposals/ballots per layer. 
- hare preround message
- LayerData for sync
- number of rewards in the block